packages/core/src/utilities/pointInShapeCallback.ts

```diff
@@ -0,0 +1,179 @@
+import { vec3 } from 'gl-matrix';
+import type { vtkImageData } from '@kitware/vtk.js/Common/DataModel/ImageData';
+import type BoundsIJK from '../types/BoundsIJK';
+import type { CPUImageData, Point3 } from '../types';
+
+export type PointInShape = {
+  value: number;
+  index: number;
+  pointIJK: vec3;
+  pointLPS: vec3 | number[];
+};
+
+export type PointInShapeCallback = ({
+  value,
+  index,
+  pointIJK,
+  pointLPS,
+}: {
+  value: number;
+  index: number;
+  pointIJK: Point3;
+  pointLPS: Point3;
+}) => void;
+
+export type ShapeFnCriteria = (pointLPS: vec3, pointIJK: vec3) => boolean;
+
+/**
+ * @deprecated
+ * You should use the voxelManager.forEach method instead.
+ * This method is deprecated and will be removed in a future version.
+ *
+ * For each point in the image (If boundsIJK is not provided, otherwise, for each
+ * point in the provided bounding box), It runs the provided callback IF the point
+ * passes the provided criteria to be inside the shape (which is defined by the
+ * provided pointInShapeFn)
+ *
+ * @param imageData - The image data object.
+ * @param dimensions - The dimensions of the image.
+ * @param pointInShapeFn - A function that takes a point in LPS space and returns
+ * true if the point is in the shape and false if it is not.
+ * @param callback - A function that will be called for
+ * every point in the shape.
+ * @param boundsIJK - The bounds of the volume in IJK coordinates.
+ */
+export function pointInShapeCallback(
+  imageData: vtkImageData | CPUImageData,
+  pointInShapeFn: ShapeFnCriteria,
+  callback?: PointInShapeCallback,
+  boundsIJK?: BoundsIJK
```

Comment:
Can you change to an options argument here, that allows specifying boundsIJK, but also other types of ranges/sets of values for future calculations?  For example, knowing an orthogonal vector and a set of min/max basis points with a min/max generator function for each row allows an optimized sphere function to be calculated.  The pointInShapeFn should be part of hte options, and it should indicate whether or not to return the set of points.   There are just a lot of optimizations one can make for this which rely on having alternate options for how to determine the content.